frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -300,48 +364,178 @@ function ReaderBookmark:onShowBookmark()
+            end,
+        })
+        table.insert(first_button_row, {
+            text = _("Remove"),
+            callback = function()
+                UIManager:show(ConfirmBox:new {
+                    text = _("Do you want remove this bookmark?"),
+                    cancel_text = _("Cancel"),
+                    cancel_callback = function()
+                        return
+                    end,
+                    ok_text = _("Remove"),
+                    ok_callback = function()
+                        local org_item = bookmark.bookmarks[item.index]
+                        bookmark:removeHighlight(org_item)
+                        UIManager:close(self.textviewer)
+                        if #bookmark.bookmarks > 1 then
+                            bm_menu:switchItemTable(nil, bookmark.bookmarks, -1)
+                            -- if no bookmarks left after removal, return to reader:
+                        else
+                            return false
+                        end
+                    end,
+                })
+            end,
+        })
+        local second_button_row = {
+            {
+                text = "<<",
+                callback = function()
+                    UIManager:close(self.textviewer)
+                    bm_menu:onMenuHold(bookmarks[1])
+                end,
+            },
+            {
+                text = "<",
+                callback = function()
+                    UIManager:close(self.textviewer)
+                    local prev = current_bookmark - 1
+                    if prev < 1 then
+                        prev = #bookmarks
+                    end
+                    bm_menu:onMenuHold(bookmarks[prev])
+                end,
+            },
+            {
+                text = _("Close"),
+                is_enter_default = true,
+                callback = function()
+                    UIManager:close(self.textviewer)
+                    -- Also close list of bookmarks, for quick return to document. To only close the textviewer, use the close button in the upper right corner:
+                    UIManager:close(bookmark.bookmark_menu)
+                end,
+            },
+            {
+                text = ">",
+                callback = function()
+                    UIManager:close(self.textviewer)
+                    local next = current_bookmark + 1
+                    if next > #bookmarks then
+                        next = 1
+                    end
+                    bm_menu:onMenuHold(bookmarks[next])
+                end,
+            },
+            {
+                text = ">>",
+                callback = function()
+                    UIManager:close(self.textviewer)
+                    bm_menu:onMenuHold(bookmarks[#bookmarks])
+                end,
+            },
+        }
+        -- if there is only one bookmark, we need only to retain the close button:
+        if #bookmark.bookmarks == 1 then
```

Comment:
For consistency, better to have always the same buttons. Just may be make some of them disabled with `enabled = #bookmark.bookmarks > 1`

Reply:
Would you say the same about the navigation buttons on the first row (for navigating found terms)?

Reply:
Just a quick comment from my phone, so grain of salt & all (may not apply), but generally speaking consistency with grayed out is better for expectations/user training than forcing reanalysis.